query-engine/connector-test-kit-rs/query-engine-tests/tests/queries/filters/many_relation.rs

```diff
@@ -514,6 +514,182 @@ mod many_relation {
+            identity    Identity  @relation(fields: [identityId], references: [id])
+          }
+
+          model Audience {
+            #id(id, String, @id)
+            deletedAt  DateTime?
+            subscriptions Subscription[]
+          }"#
+        };
+
+        schema.to_owned()
+    }
+
+    // Regression test for https://github.com/prisma/prisma/issues/25103
+    // SQL Server excluded because the m2m fragment does not support onUpdate/onDelete args which are needed.
+    #[connector_test(schema(schema_25103), exclude(SqlServer))]
+    async fn prisma_25103(runner: Runner) -> TestResult<()> {
+        // Create some sample audiences
+        run_query!(
+            &runner,
+            r#"mutation {
+              createOneAudience(data: {
+                id: "audience1",
+                deletedAt: null
+              }) {
+                id
+            }}"#
+        );
+        run_query!(
+            &runner,
+            r#"mutation {
+              createOneAudience(data: {
+                id: "audience2",
+                deletedAt: null
+              }) {
+                id
+            }}"#
+        );
+
+        // Create a contact with identities and subscriptions
+        insta::assert_snapshot!(
+          run_query!(
+              &runner,
+              r#"mutation {
+              createOneContact(data: {
+                id: "contact1",
+                identities: {
+                  create: [
+                    {
+                      id: "identity1",
+                      subscriptions: {
+                        create: [
+                          {
+                            id: "subscription1",
+                            audienceId: "audience1",
+                            optedOutAt: null
+                          },
+                          {
+                            id: "subscription2",
+                            audienceId: "audience2",
+                            optedOutAt: null
+                          }
+                        ]
+                      }
+                    }
+                  ]
+                }
+              }) {
+                id,
+                identities (orderBy: { id: asc }) {
+                  id,
+                  subscriptions (orderBy: { id: asc }) {
+                    id,
+                    audienceId
+                  }
+                }
+            }}"#
+          ),
+          @r###"{"data":{"createOneContact":{"id":"contact1","identities":[{"id":"identity1","subscriptions":[{"id":"subscription1","audienceId":"audience1"},{"id":"subscription2","audienceId":"audience2"}]}]}}}"###
+        );
+
+        insta::assert_snapshot!(
+            run_query!(
+                &runner,
+                r#"query {
+                    findManyContact {
+                        identities {
+                            subscriptions(where: { audience: { deletedAt: { equals: null } } }) {
```

Comment:
In particular, this line misses a filter:

```suggestion
                            subscriptions(where: { audience: { deletedAt: { equals: null } } }) {
                            subscriptions(orderBy: { id: asc }, where: { optedOutAt: null, audience: { deletedAt: null } }) {
```

Reply:
fwiw the missing `optedOutAt` filter is irrelevant to the issue, it's `deletedAt` which is the problem (and is present here already)

it's nice to have tests with more complex queries though